src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -0,0 +1,316 @@
+            writer.write(securityConfig);
+        } catch (IOException e) {
+            System.err.println("Exception writing security configuration to opensearch.yml.");
+            System.exit(-1);
+        }
+    }
+
+    /**
+     * Helper method to build security configuration to append to opensearch.yml
+     * @return the configuration string to be written to opensearch.yml
+     */
+    static String buildSecurityConfigString() {
+        StringBuilder securityConfigLines = new StringBuilder();
+
+        securityConfigLines.append("\n")
+            .append("######## Start OpenSearch Security Demo Configuration ########\n")
+            .append("# WARNING: revise all the lines below before you go into production\n")
+            .append("plugins.security.ssl.transport.pemcert_filepath: esnode.pem\n")
+            .append("plugins.security.ssl.transport.pemkey_filepath: esnode-key.pem\n")
+            .append("plugins.security.ssl.transport.pemtrustedcas_filepath: root-ca.pem\n")
+            .append("plugins.security.ssl.transport.enforce_hostname_verification: false\n")
+            .append("plugins.security.ssl.http.enabled: true\n")
+            .append("plugins.security.ssl.http.pemcert_filepath: esnode.pem\n")
+            .append("plugins.security.ssl.http.pemkey_filepath: esnode-key.pem\n")
+            .append("plugins.security.ssl.http.pemtrustedcas_filepath: root-ca.pem\n")
+            .append("plugins.security.allow_unsafe_democertificates: true\n");
+
+        if (initsecurity) {
+            securityConfigLines.append("plugins.security.allow_default_init_securityindex: true\n");
+        }
+
+        securityConfigLines.append("plugins.security.authcz.admin_dn:\n  - CN=kirk,OU=client,O=client,L=test, C=de\n\n");
+
+        securityConfigLines.append("plugins.security.system_indices.enabled: true\n" + "plugins.security.system_indices.indices: [")
+            .append(SYSTEM_INDICES)
+            .append("]\n");
+
+        if (!isNetworkHostAlreadyPresent(OPENSEARCH_CONF_FILE)) {
+            if (cluster_mode) {
+                securityConfigLines.append("network.host: 0.0.0.0\n");
+                securityConfigLines.append("node.name: smoketestnode\n");
+                securityConfigLines.append("cluster.initial_cluster_manager_nodes: smoketestnode\n");
+            }
+        }
+
+        if (!isNodeMaxLocalStorageNodesAlreadyPresent(OPENSEARCH_CONF_FILE)) {
+            securityConfigLines.append("node.max_local_storage_nodes: 3\n");
+        }
+
+        securityConfigLines.append("######## End OpenSearch Security Demo Configuration ########\n");
+
+        return securityConfigLines.toString();
+    }
+
+    /**
+     * Helper method to check if network.host config is present
+     * @param filePath path to opensearch.yml
+     * @return true is present, false otherwise
+     */
+    static boolean isNetworkHostAlreadyPresent(String filePath) {
+        try {
+            String searchString = "^network.host";
```

Comment:
Just for my knowledge, is the `^` sign escaping of the comment? I know that we had this line as a default comment in the config file.

Reply:
Ah, thats good to know.